gtsam/nonlinear/NonlinearFactor.h

```diff
@@ -272,503 +307,306 @@ class GTSAM_EXPORT NoiseModelFactor: public NonlinearFactor {
 
 
 /* ************************************************************************* */
+/* We need some helper structs to help us with NoiseModelFactorN - specifically
```

Comment:
A great exercise in meta-programming which I’d love to see disappear, see comment below :-)